### PR TITLE
Project support

### PR DESCRIPTION
**Pull request by anthmatic**
## Description and Context
Updated project support:

`projects` renamed to `project`

- [x] hs project init
- [x] hs project upload

Changes to `hsproject.json`:
```json
{
  "name": "weather_app",
  "srcDir": "src"
}
```

NOTE: I've purposely left some TODOs that I'll address later, I'd just like to get the basics in so that we can get a solid feedback loop in.

### CODE REVIEW

File: packages/cli/commands/projects/init.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  logger.log('Use `hs project --help` to learn more about the command.');
+  logger.log('-------------------------------------------------------------');
+};
+
+exports.builder = yargs => {
+  yargs.positional('path', {
+    describe: 'Path to a project folder',
+    type: 'string',
+  });
+  // TODO:
```
Comment (anthmatic) [inline packages/cli/commands/projects/init.js:94]: I was previously accounting for these to be passed as options. Not exactly sure how that should work with the new config file. Thinking of keeping these and getting some UX feedback.

File: packages/cli/commands/projects.js [hunk 2]
```diff
@@ -4,16 +4,21 @@ const {
...
+exports.command = 'project';
 exports.describe = false; //'Commands for working with projects';
 
 exports.builder = yargs => {
   addOverwriteOptions(yargs, true);
   addConfigOptions(yargs, true);
   addAccountOptions(yargs, true);
 
+  // TODO: deploy must be updated
   yargs.command(deploy).demandCommand(1, '');
```
Comment (anthmatic) [inline packages/cli/commands/projects.js:19]: Not currently used in the flow. Will need to be adjusted eventually, but will mostly stay the same so keeping this for now.

File: packages/cli/commands/projects/init.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
+const {
+  addAccountOptions,
+  addConfigOptions,
+  setLogLevel,
+  getAccountId,
+  addUseEnvironmentOptions,
+} = require('../../lib/commonOpts');
+// const { trackCommandUsage } = require('../../lib/usageTracking');
```
Comment (miketalley) [inline packages/cli/commands/projects/init.js:8]: Should we add some tracking?

File: packages/cli-lib/api/dfs.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+ *
+ * @async
+ * @returns {Promise}
+ */
+async function fetchProjects(portalId) {
+  return http.get(portalId, {
+    uri: DEVELOPER_FILE_SYSTEM_API_PATH,
+  });
+}
+// TODO: paging?
```
Comment (miketalley) [inline packages/cli-lib/api/dfs.js:17]: Did we want to add paging, or is this a reminder for later?

File: packages/cli/commands/projects/init.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+exports.command = 'init [path]';
+exports.describe = false;
+
+exports.handler = async options => {
+  loadAndValidateOptions(options);
+
+  const { path: projectPath } = options;
+  const accountId = getAccountId(options);
+
+  // TODO:
```
Comment (miketalley) [inline packages/cli/commands/projects/init.js:47]: Should we enable this now?

File: packages/cli/commands/projects/init.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+  logger.log('1. hs project upload');
+  logger.log(
+    '   Upload your project files to HubSpot. Upload action adds your files to a build.'
+  );
+  logger.log();
+  logger.log('2. View your changes on the preview build url');
+  logger.log();
+  logger.log('Use `hs project --help` to learn more about the command.');
+  logger.log('-------------------------------------------------------------');
+};
```
Comment (miketalley) [inline packages/cli/commands/projects/init.js:87]: 👍

**Comment by brandenrodgers** [comment]
Pending answers to Mike's questions this all lgtm! This is a solid first pass 👌

**Review by miketalley — COMMENTED** [review]
LGTM -- Just wanted to check if we should address the TODOs now or if they are meant to be left for later.

File: packages/cli-lib/api/dfs.js (hunk 4 above)
Comment (anthmatic) [reply packages/cli-lib/api/dfs.js:17]: We'll need to add paging but I'll remove the TODO for now

**Comment by anthmatic** [comment]
> if we should address the TODOs now or if they are meant to be left for later

These were mostly reminders for me to come back to later or enable things like tracking when we really need it but in the interest of keeping things cleaner I enabled tracking and removed most of the TODOs for now.

**Pull request merged**